contracts/SpokePool.sol

```diff
@@ -225,49 +222,130 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
         });
         bytes32 relayHash = _getRelayHash(relayData);
 
+        _fillRelay(relayHash, relayData, relayerFeePct, maxTokensToSend, repaymentChain);
+    }
+
+    function _fillRelay(
+        bytes32 relayHash,
+        RelayData memory relayData,
+        uint64 updatableRelayerFeePct,
+        uint256 maxTokensToSend,
+        uint256 repaymentChain
+    ) internal {
+        // We limit the relay fees to prevent the user spending all their funds on fees.
+        require(
+            updatableRelayerFeePct <= 0.5e18 &&
```

Comment:
I think one of these checks is redundant. if you validate `relayData.realizedLpFeePct` and `(updatableRelayerFeePct + relayData.realizedLpFeePct) < 1e18` then it's impossible for `updatableRelayerFeePct` to be `>0.5e18`.

Reply:
what if `realizedLpFeePct = 0` but `updatableRelayerFeePct = 0.6`?

Reply:
then just check that: `updatableRelayerFeePct <= 0.5e18 && relayData.realizedLpFeePct <= 0.5e18`?

Reply:
true